grove/alpha/amplification/oracles.py

```diff
@@ -0,0 +1,48 @@
+"""Module containing implementations of oracle Programs.
+"""
+
+import numpy as np
+import pyquil.quil as pq
+from pyquil.gates import X, Z
+
+from grove.alpha.utility_programs import ControlledProgramBuilder
+
+
+def basis_selector_oracle(qubits, bitstring):
+    """Defines an oracle that selects the ith element of the computational basis.
+
+    Flips the sign of the state :math:`\\vert x\\rangle>`
+    if and only if x==bitstring and does nothing otherwise.
+
+    :param qubits: The qubits the oracle is called on. The qubits are assumed to be ordered from
+     most significant qubit to least significant qubit.
+    :param bitstring: The desired bitstring, given as a string of ones and zeros. e.g. "101"
+    :return: A program representing this oracle.
+    :rtype: Program
+    """
+    if len(qubits) != len(bitstring):
+        raise ValueError(
+            "The bitstring should be the same length as the number of qubits.")
```

Comment:
fits on a single line

Reply:
Did you pull this or do you have good eyes?